components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java

```diff
@@ -316,5 +319,54 @@ public void updateRoleName(int tenantId, String oldRoleName, String newRoleName)
             throw new IdentitySCIMException("Error when updating role name of the role: " + oldRoleName);
         }
     }
+    /**
+     * Lists the Groups created from SCIM with a attribute filter and search regex
+     *
+     * @return list of SCIM groups
+     * @throws IdentitySCIMException
+     */
+    public String[] getGroupNameList(String searchAttributeName, String searchAttributeValue, Integer tenantId)
+            throws IdentitySCIMException {
+
+        List<String> roleList = new ArrayList<>();
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection()) {
+
+            try (PreparedStatement prepStmt =
+                         connection.prepareStatement(SQLQueries.LIST_SCIM_GROUPS_SQL_BY_ATT_AND_ATT_VALUE)) {
+
+                prepStmt.setInt(1, tenantId);
+                prepStmt.setString(2, searchAttributeName);
+                prepStmt.setString(3, searchAttributeValue);
+
+                try (ResultSet rSet = prepStmt.executeQuery()) {
+                    while (rSet.next()) {
+                        String roleName = rSet.getString(1);
+                        if (StringUtils.isNotEmpty(roleName)) {
+                            if (roleName.contains(CarbonConstants.DOMAIN_SEPARATOR)) {
+                                String[] parts = roleName.split(CarbonConstants.DOMAIN_SEPARATOR);
+                                roleList.add(parts[parts.length - 1]);
+                            } else {
+                                roleList.add(roleName);
+                            }
+                        }
+                    }
+                } catch (SQLException e) {
```

Comment:
having only one catch is enough for these 3 try with resources statements.

Reply:
done